_posts/2023-11-20-securing-application-with-other-oidc.adoc

```diff
@@ -0,0 +1,274 @@
+. Under the *Add an app to use Single Sign-on* option, click on *Add App* to add the application we want to secure. Click on *Create New App*, choose *OIDC - OpenID Connect* under `Sign-in method` and *Web Application* for `Application type` and click `Next`. 
+. Under *General Settings* add the following configurations: 
+* `App integration name:` *simple-webapp-okta*
+* `Grant type:` Check *Client Credentials* 
+* `Sign-in redirect URIs`: Leave it as is for now. We will edit it later.
+* Under the `Assignments` tab, for `Controlled access`, select *Allow everyone in your organization to access* and disable *Federation Broker Mode*. 
+* Click `Save`
+. Navigate to the Directory drop down on the left panel and click `People`. Click *Add person* and add a person with: 
+* First name: *Alice*
+* Last name: *Smith*
+* Email: *alice@example.org*
+* User Name: *alice@example.org* 
+* Check `I will set password` and set a password that fits the password requirements. 
+* Check off `User must change password on first login` and hit `Save`.
+. Go back to your application using the `Application` tab on the left hand panel and under assignments, click *Assign* and *Assign to People* and select *Alice*. For more information about how to assign a user to an application, please see https://support.okta.com/help/s/article/How-To-Assign-An-User-To-An-Application?language=en_US[here] and to learn more about how to add a user manually https://help.okta.com/en-us/content/topics/users-groups-profiles/usgp-add-users.htm[click here].
+
+=== Extracting Provider Url
+In order to extract the user information through the id token, we will need to configure a custom authorization server. Follow the steps below for this:
+
+. Under the *Security* dropdown, select *API* and click on the edit button next to the *Default* server. 
+. Under the *Claims* tab, click on the *Add Claim* button and to add a claim with the following configurations: 
+* Name: *IDTClaim*
+* Include in token type: *ID Token* and keep it at always. 
+* Value type: *Expression*
+* Value: *appuser.userName* and hit `Save`. 
+You can learn more about Expression Language Reference for Okta https://developer.okta.com/reference/okta_expression_language/[here].
+. Now go back to the API menu and copy the *Issuer URI* for the default server. 
+. This will be used as the provider url when configuring our WildFly application.
+
+== Add Helm Configuration
+. Keep the URL obtained from the last step. 
+. Switch to the charts directory in the simple-webapp-okta example.
+```
+cd /PATH/TO/ELYTRON/EXAMPLES/simple-webapp-okta/charts
+```
+Notice there’s a helm.yaml file in this directory with the following content:
+```
+build:
+  uri: https://github.com/wildfly-security-incubator/elytron-examples.git       // <1>
+  ref: <branch_name>        // <2>
+  contextDir: simple-webapp-okta
+deploy:
+  replicas: 1
+  env:
+    - name: OIDC_PROVIDER_URL
+      value: <okta_URL>     // <3>
```

Comment:
The client ID and client secret could also be specified here as environment variables.

Reply:
updated instructions and application to use env.OIDC_CLIENT_ID and env.OIDC_CLIENT_SECRET